solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java

```diff
@@ -1306,7 +1306,7 @@ private static void forceLeaderElection(SolrQueryRequest req, CollectionsHandler
     try (ZkShardTerms zkShardTerms = new ZkShardTerms(collectionName, slice.getName(), zkController.getZkClient())) {
       // if an active replica is the leader, then all is fine already
       Replica leader = slice.getLeader();
-      if (leader != null && leader.getState() == State.ACTIVE) {
+      if (leader != null && leader.getState() == State.ACTIVE && zkShardTerms.getHighestTerm() == zkShardTerms.getTerm(leader.getName())) {
```

Comment:
I know this is not new code, but should we change `leader.getState() == State.ACTIVE` to `leader.isActive(liveNodes)`?

Reply:
I think I need to revert this part since I'm not touching terms anymore anyway.